src/common/packager.ts

```diff
@@ -61,20 +62,23 @@ export class Packager {
                 this.awaitStart())
             .then(() => {
                 if (executedStartPackagerCmd) {
-                    Log.logMessage("Packager started.", outputChannel);
+                    Log.logMessage("Packager started.");
                 } else {
-                    Log.logMessage("Packager is already running.", outputChannel);
+                    Log.logMessage("Packager is already running.");
                     if (!this.packagerProcess) {
-                        Log.logWarning("Debugging is not supported if the React Native Packager is not started within VS Code. "
-                            + "If debugging fails, please kill other active React Native packager processes and retry.", null, outputChannel);
+                        Log.logWarning(ErrorHelper.getWarning("Debugging is not supported if the React Native Packager is not started within VS Code. If debugging fails, please kill other active React Native packager processes and retry."));
                     }
                 }
             });
     }
 
-    public stop(outputChannel?: OutputChannel): Q.Promise<void> {
-        return new CommandExecutor(this.projectPath).killReactPackager(this.packagerProcess, outputChannel).then(() =>
-            this.packagerProcess = null);
+    public stop(): Q.Promise<void> {
+        if (!this.packagerProcess && this.isRunning()) {
```

Comment:
this.isRunning() returns a promise, so I think right now the result will be ignored and we're just checking the non-nullness of the returned promise which should always be truthy. If we want to check if the packager is running then we'll need to wait for the promise to resolve